packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -222,21 +235,26 @@
           }
         }
       );
-      this.emitQueryPreviewRequestUpdated(this.queryPreviewRequest);
+
+      // If the query has been saved it will emit load instead of the emitting the updated request.
+      if (this.isSavedQuery && this.saveCache) {
+        this.$emit('load', this.queryPreviewInfo.query);
```

Comment:
Not sure if this is right. If the query preview errored in the first request, and it was saved with an error status in the cache, you are now emitting an event saying it's all good, which is not correct.

You have to check the status of the query preview in the cache too.